spinta/cli/helpers/sync/helpers.py

```diff
@@ -121,27 +124,25 @@ def get_data_service_name_prefix(credentials: RemoteClientCredentials) -> str:
 
 
 def clean_private_attributes(resource: Resource) -> dict[str, Model]:
-    resource.external = ""
+    # Resource cleaning
+    for field in resource.__annotations__:
+        if field not in EXCLUDED_RESOURCE_FIELDS:
+            setattr(resource, field, "")
     resource.type = ""
-    resource.backend.name = ""
 
-    resource_models = resource.models
-    for model_name, model in resource_models.items():
-        for property_name, property in model.properties.items():
-            if property.basename.startswith("_"):  # Skip
-                continue
-            if property.visibility not in {Visibility.private, None}:
-                continue
-            property.external = ""
+    # Model cleaning
+    resource.models = {name: model for name, model in resource.models.items() if model.visibility != Visibility.private}
 
-        if model.visibility not in {Visibility.private, None}:
-            continue
-        model.external.name = ""  # Source column cleanup.
+    # Property cleaning
+    for model in resource.models.values():
+        model.properties = {
+            name: property for name, property in model.properties.items() if property.visibility != Visibility.private
+        }
```

Comment:
According to best practices comments should not say **what** code is doing, but "why". If you think that code is not self explanatory enough - refactor this code part to separate function.